packages/react-components/react-nav-preview/library/src/components/Nav/useNav.ts

```diff
@@ -113,12 +106,22 @@ export const useNav_unstable = (props: NavProps, ref: React.Ref<HTMLDivElement>)
     currentSelectedCategoryValue.current = selectedCategoryValue;
```

Comment:
I think we should just set the refs as part of the render function, this will ensure anything that relies on getRegisteredNavItems during render will always be up to date with latest render props, it should be safe from a perf standpoint.

![image](https://github.com/user-attachments/assets/e0c8d70d-6889-4586-af5a-e551083eb6fa)